cpp/src/Ice/StreamSocket.cpp

```diff
@@ -321,6 +336,24 @@ StreamSocket::write(const char* buf, size_t length)
     return sent;
 }
 
+bool
+StreamSocket::isWaitingToBeRead() const noexcept
+{
+#if defined(ICE_USE_IOCP)
+    if (hasBytesAvailable(_fd))
```

Comment:
Most likely because there's already read data buffered and there's no need to start an asynchronous read on the underlying tcp/ssl transceiver. That's the case when `preRead` returns false here https://github.com/zeroc-ice/ice/blob/05fbaef93520dd07307a8abb7d5143ef935e2e23/cpp/src/Ice/WSTransceiver.cpp#L702

In this case, we just trigger a read event on the completion port. In turn, the thread pool will call right away `finishRead`.